### PR TITLE
REF: implement groupby std, min, max as EA methods

### PR DESCRIPTION
**Pull request by jbrockmendel**
xref #51003 does the same thing for quantile
xref #43682 asking for something similar (but not for these particular reductions)

Allows EA authors to implement performant implementations (including us for e.g. pyarrow dtypes).

### CODE REVIEW

**Review by rhshadrach — COMMENTED** [review]
There any performance impact here?

File: pandas/core/arrays/masked.py [hunk 1]
```diff
@@ -1383,3 +1384,64 @@ def _accumulate(
...
+        self,
+        *,
+        ngroups: int,
+        ids: npt.NDArray[np.intp],
+        val_test: Literal["any", "all"],
+        skipna: bool,
+    ):
+        from pandas.core.arrays import BooleanArray
+
+        result = np.zeros(ngroups * 1, dtype=np.int8).reshape(-1, 1)
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/arrays/masked.py:1401]: probably leftover from cut/paste/simplify

**Comment by jbrockmendel** [comment]
> There any performance impact here?

Not directly, but a big part of the idea is to make it easier to implement performant implementations for our pyarrow arrays (and in my daydreams, for 3rd party authors to do Distributed/GPU EA implementations)

**Review by rhshadrach — APPROVED** [review]
Both the direction this moves toward and the implementation look good to me. cc @mroeschke @jorisvandenbossche

**Comment by mroeschke** [comment]
Does any/all/std for sure need a different API then `groupby_op` in https://github.com/pandas-dev/pandas/pull/51166?

**Comment by jbrockmendel** [comment]
> Does any/all/std for sure need a different API then groupby_op in https://github.com/pandas-dev/pandas/pull/51166?

No.  #52053 moves std to use WrappedCythonOp, so it would no longer need a separate method here.  #52043 is a step in the same direction for any/all

**Comment by mroeschke** [comment]
Okay so IIUC, `_groupby_any_all` and `_groupby_std` is a bridge to `_groupby_op` in https://github.com/pandas-dev/pandas/pull/51166?

**Comment by jbrockmendel** [comment]
> Okay so IIUC, _groupby_any_all and _groupby_std is a bridge to _groupby_op in https://github.com/pandas-dev/pandas/pull/51166?

Yes.  Though at this point I'm _almost_ ready to close this.

**Comment by jbrockmendel** [comment]
Closing in favor of #52089

**Pull request closed without merging**